lib/collection/src/wal.rs

```diff
@@ -92,8 +92,8 @@ impl<'s, R: DeserializeOwned + Serialize + Debug> SerdeWal<R> {
 
         (start_from..(first_index + num_entries)).map(move |idx| {
             let record_bin = self.wal.entry(idx).expect("Can't read entry from WAL");
-            let record: R = serde_cbor::from_slice(&record_bin.to_vec())
-                .or_else(|_err| rmp_serde::from_slice(&record_bin.to_vec()))
+            let record: R = serde_cbor::from_slice(&record_bin)
```

Comment:
For history, this is possible because `Entry` implements `Deref` this way.

https://github.com/qdrant/wal/blob/0dd3943113ff7ec2fbc5428bb77ba206c8492fa9/src/segment.rs#L35-L40